File: rl-designer/src/services/explorer.ts
import { invoke } from '@tauri-apps/api/core';
import type { DecalTextures } from '@/types';
import type { ElementType } from '@/constants/elements';

interface GitHubDecalsResponse {
    success: boolean;
    decals: DecalTextures[];
    error?: string;
}

export const getDecalsFromGitHub = async (element: ElementType): Promise<GitHubDecalsResponse> => {
    try {
        const result = await invoke<GitHubDecalsResponse>(`get_element_decals_from_github`, { element });
        return result;
    } catch (error) {
        console.error(`Failed to fetch ${element} decals from GitHub:`, error);
        return {
            success: false,
            decals: [],
            error: `Failed to invoke command: ${error}`
        };
    }
};

File: rl-designer/src-tauri/src/commands/explorer.rs
use crate::types::decal::DecalInfo;
use crate::types::elements::ElementType;
use crate::utils::explorer::fetch_decals_from_github_raw;

#[derive(Debug, serde::Serialize)]
pub struct GitHubDecalsResponse {
    success: bool,
    decals: Vec<DecalInfo>,
    error: Option<String>,
}

#[tauri::command]
pub async fn get_element_decals_from_github(element: ElementType) -> GitHubDecalsResponse {
    match fetch_decals_from_github_raw(element).await {
        Ok(decals) => GitHubDecalsResponse {
            success: true,
            decals,
            error: None,
        },
        Err(e) => GitHubDecalsResponse {
            success: false,
            decals: Vec::new(),
            error: Some(e),
        },
    }
}


